pkg/autoupdate/tuf/library_manager.go

```diff
@@ -121,7 +122,8 @@ func (ulm *updateLibraryManager) stageAndVerifyUpdate(binary autoupdatableBinary
 	stagedUpdatePath := filepath.Join(ulm.stagingDir, targetFilename)
 
 	// Request download from mirror
-	resp, err := ulm.mirrorClient.Get(ulm.mirrorUrl + fmt.Sprintf("/kolide/%s/%s/%s", binary, runtime.GOOS, targetFilename))
+	downloadPath := path.Join("/", "kolide", string(binary), runtime.GOOS, PlatformArch(), targetFilename)
+	resp, err := ulm.mirrorClient.Get(ulm.mirrorUrl + downloadPath)
```

Comment:
`path.Join(ulm.mirrorUrl, "kolide", ...)` and you don't need the leading `/`, right?

Reply:
Unfortunately looks like this doesn't work because path.Join will strip the second / out of the scheme: https://play.golang.com/p/qD-wPK4x6Zw 😩

Reply:
Oh.. This is URL constructions. I bet there's a URL helper. But this seems fine